packages/safe-tools-client/app/components/safe-info/index.gts

```diff
@@ -38,16 +53,36 @@ export default class SafeInfo extends Component<Signature> {
     {{#if @isLoadingSafes}}
       <div class='info'>Loading safes...</div>
     {{else if (gt @safes.length 1)}}
-      <BoxelSelect
-        class='safe-tools__dashboard-dropdown'
-        @selected={{@currentSafe}}
-        @onChange={{@onSelectSafe}}
-        @options={{this.safes}}
-        @dropdownClass="boxel-select-usage-dropdown"
-        data-test-safe-dropdown
-        as |safe itemCssClass|>
-        <div class="{{itemCssClass}} blockchain-address">{{truncateMiddle safe.address}}</div>
-      </BoxelSelect>
+      <div class='safe-tools_safe-info-address'>
+        <BoxelSelect
+          class='safe-tools__dashboard-dropdown'
+          @selected={{@currentSafe}}
+          @onChange={{@onSelectSafe}}
+          @options={{this.safes}}
+          @dropdownClass="boxel-select-usage-dropdown"
+          data-test-safe-dropdown
+          as |safe itemCssClass|>
+          <div class="{{itemCssClass}} blockchain-address">{{truncateMiddle safe.address}}</div>
+        </BoxelSelect>
+        <BoxelIconButton
+          @icon="copy"
+          aria-label="Copy to Clipboard"
+          style={{cssVar
+              boxel-icon-button-width=0
+              boxel-icon-button-height="100%"
+              icon-color="var(--boxel-light)"
+            }}
+          {{on "click"
+                (copyToClipboard
+                  value=@currentSafe.address
+                  onCopy=this.flashCopiedConfirmation
+                )
+              }}
+        />
```

Comment:
It is a little weird from an information architecture perspective to have this outside of the dropdown, but I think given the options it is OK.